src/modules/Thread/context/hooks/useSendFileMessageV2.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback } from 'react';
+import { GroupChannel } from '@sendbird/chat/groupChannel';
+import { FileMessage, FileMessageCreateParams } from '@sendbird/chat/message';
+
+import { CustomUseReducerDispatcher, Logger } from '../../../../lib/SendbirdState';
+import { ThreadContextActionTypes } from '../dux/actionTypes';
+import topics, { SBUGlobalPubSub } from '../../../../lib/pubSub/topics';
+import { scrollIntoLast } from '../utils';
+import { SendableMessageType } from '../../../../utils';
+import { PublishingModuleType } from './useSendMultipleFilesMessage';
+import { SCROLL_BOTTOM_DELAY_FOR_SEND } from '../../../../utils/consts';
+import { SBUEventHandlers } from '../../../../lib/types';
+
+interface DynamicProps {
+  currentChannel: GroupChannel;
+  onBeforeSendFileMessage?: (file: File, quotedMessage?: SendableMessageType) => FileMessageCreateParams;
+}
+interface StaticProps {
+  logger: Logger;
+  pubSub: SBUGlobalPubSub;
+  threadDispatcher: CustomUseReducerDispatcher;
+  eventHandlers?: SBUEventHandlers
+}
+
+interface LocalFileMessage extends FileMessage {
+  localUrl: string;
+  file: File;
+}
+
+export type SendFileMessageFunctionTypeV2 = (params: FileMessageCreateParams, quotedMessage?: SendableMessageType) => Promise<FileMessage>;
+
+export default function useSendFileMessageCallbackV2({
+  currentChannel,
+  onBeforeSendFileMessage,
+}: DynamicProps, {
+  logger,
+  eventHandlers,
+  pubSub,
+  threadDispatcher,
+}: StaticProps): SendFileMessageFunctionTypeV2 {
+  const sendMessage = useCallback((params: FileMessageCreateParams, quoteMessage): Promise<FileMessage> => {
+    return new Promise((resolve, reject) => {
+        if (quoteMessage) {
+          params.isReplyToChannel = true;
+          params.parentMessageId = quoteMessage.messageId;
+        }
```

Comment:
this is basically the only change in the copying over

Reply:
Can you add a `// fork note:` here?

Reply:
oh ya but there shouldn't ever be conflicts with this because its a new file. i'll add it in ThreadProvider tho!